panther_analysis_tool/testing.py

```diff
@@ -164,19 +164,50 @@ def _get_generic_error_details(self) -> Tuple[Optional[Exception], Optional[str]
             generic_error = self._detection_result.setup_exception
         return generic_error, generic_error_title
 
+    def _check_exception_types(
+        self, ignore_exception_types: Optional[List[Type[Exception]]] = None
+    ) -> None:
+        for exception_type in ignore_exception_types or []:
+            if isinstance(self._detection_result.detection_exception, exception_type):
+                self._detection_result.detection_exception = None
+            if isinstance(self._detection_result.title_exception, exception_type):
+                self._detection_result.title_exception = None
+            if isinstance(self._detection_result.description_exception, exception_type):
+                self._detection_result.description_exception = None
+            if isinstance(self._detection_result.reference_exception, exception_type):
+                self._detection_result.reference_exception = None
+            if isinstance(self._detection_result.severity_exception, exception_type):
+                self._detection_result.severity_exception = None
+            if isinstance(self._detection_result.runbook_exception, exception_type):
+                self._detection_result.runbook_exception = None
+            if isinstance(self._detection_result.destinations_exception, exception_type):
+                self._detection_result.destinations_exception = None
+            if isinstance(self._detection_result.dedup_exception, exception_type):
+                self._detection_result.dedup_exception = None
+            if isinstance(self._detection_result.alert_context_exception, exception_type):
+                self._detection_result.alert_context_exception = None
+
     def _get_detection_alert_value(self) -> bool:
-        if self._detection_result.detection_type == TYPE_POLICY:
+        if self._detection_result.detection_type.upper() == TYPE_POLICY.upper():
             return Policy.matcher_alert_value
         return Rule.matcher_alert_value
 
-    def interpret(self) -> TestResult:
+    def interpret(
+        self, ignore_exception_types: Optional[List[Type[Exception]]] = None
+    ) -> TestResult:
         """Evaluate the detection result taking into account
         the errors raised during evaluation and
         the test specification expectations"""
+
+        # first, we should update the detection result, taking into account any
+        # ignored exception types passed into this test
+        self._check_exception_types(ignore_exception_types)
```

Comment:
We need to mutate the detection result because `DetectionResult.errored` would falsely return `True` otherwise, right?

How about if we moved the function that nullifies errors to `DetectionResult`, e.g. `DetectionResult.ignore_errors(exception_types)`. Such a function is expected to mutate the detection result instance, and it is quite easier to test as well!

Reply:
> We need to mutate the detection result because DetectionResult.errored would falsely return True otherwise, right?

yea, exactly 😕